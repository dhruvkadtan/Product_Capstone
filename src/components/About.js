import "../App.css";
import carousel1 from "../Images/carousel1.jpg";
import analysis from "../Images/analysis.jpg";
import products from "../Images/products.jpg";
import signin from "../Images/signin.jpg";

const About = () => {
    return(
        
        <div
        id="carouselExampleCaptions"
        className="carousel slide"
        data-ride="carousel">
          <div className="carousel-indicators">
          <button
            type="button"
            data-bs-target="#carouselExampleCaptions"
            data-bs-slide-to="0"
            className="active"
            aria-current="true"
            aria-label="Slide 1"
          ></button>
          <button
            type="button"
            data-bs-target="#carouselExampleCaptions"
            data-bs-slide-to="1"
            aria-label="Slide 2"
          ></button>
          <button
            type="button"
            data-bs-target="#carouselExampleCaptions"
            data-bs-slide-to="2"
            aria-label="Slide 3"
          ></button>
          <button
            type="button"
            data-bs-target="#carouselExampleCaptions"
            data-bs-slide-to="3"
            aria-label="Slide 4"
          ></button>
        </div>
            <div className="carousel-inner">
                <div className="carousel-item active "> 
                <img
                    src={carousel1}
                    className="d-block  w-100"
                    alt="..."
                    style={{ filter: " blur(1px)" }}
                    />
                
                <div className="carousel-caption d-none d-md-block ">
                    <h3 className="text-white fw-bold">Welcome to D-Systems</h3>
                    <h4 className="text-white fw-bold">Accounting and Inventory Software</h4>
                        <h5>We provide a managing platform for businesses</h5>
                        <h5>   Here, we can account the stock products</h5>
                        <h5>    and also maintain daily transactions.</h5>
                </div>
                </div>
                <div className="carousel-item"> 
                <img
                    src={products}
                    className="d-block w-100"
                    alt="..."
            
                    />
                    <div className="carousel-caption d-none d-md-block">
                        <h5>Product Details Can be viewed without Sign in</h5>
                    </div>
                </div>
                <div className="carousel-item"> 
                <img
                    src={signin}
                    className="d-block w-100"
                    alt="..."
                    
                    />
                    <div className="carousel-caption d-none d-md-block">
                        <h5>To add, edit, delete Please SignIn first</h5>
                    </div>
                </div>
                <div className="carousel-item"> 
                <img
                    src={analysis}
                    className="d-block w-100"
                    alt="..."
                    
                    />
                    <div className="carousel-caption d-none d-md-block">
                        <h5>Products views and price analysis can be done through analysis</h5>
                    </div>
                </div>
                
               
            </div>
          
    </div>
    )
}

export default About;

/* <div
                id="carouselExampleCaptions"
                className="carousel slide"
                data-ride="carousel">
                    
                    <div className="carousel-inner">
                        <div className="carousel-item active "> 
                        <img
                            src={carousel1}
                            className="d-block  w-100"
                            alt="..."
                            style={{ filter: " blur(1px)" }}
                            />
                        </div>
                        <div className="carousel-caption d-none d-md-block ">
                            <h3 className="text-white fw-bold">Welcome to D-Systems</h3>
                            <h4 className="text-white fw-bold">Accounting and Inventory Software</h4>
                        </div>
                        <div className="carousel-item"> 
                        <img
                            src={carousel1}
                            className="d-block w-100"
                            alt="..."
                            style={{ filter: " blur(1px)" }}
                            />
                            <div className="carousel-caption d-none d-md-block">
                                <h4>We provide a managing platform for businesses</h4>
                                <h4>   Here, we can account the stock products</h4>
                                <h4>    and also maintain daily transactions.</h4>
                            </div>
                        </div>
                       
                    </div>
                    <a class="carousel-control-prev" href="#carouselExampleControls" role="button" data-slide="prev">
                        <span class="carousel-control-prev-icon" aria-hidden="true"></span>
                        <span class="sr-only">Previous</span>
                     </a>
                    <a class="carousel-control-next" href="#carouselExampleControls" role="button" data-slide="next">
                        <span class="carousel-control-next-icon" aria-hidden="true"></span>
                        <span class="sr-only">Next</span>
                    </a>
            </div>
    */