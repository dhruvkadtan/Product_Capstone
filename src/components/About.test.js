import React from 'react';
import renderer from 'react-test-renderer';
import  About from './About';
import { screen,render as renderRtl } from '@testing-library/react';

const setup = () => renderRtl(<About/>)

describe('App',() => {
  test('render correctly', () => {
    const tree = renderer.create(<About/>)
    expect(tree).toMatchSnapshot();
  })

})

describe("checking dom elements", () => {
  test('Correct slide1 heading', () => {
    setup();
    const heading = screen.getByText("Welcome to D-Systems")
    expect(heading).toBeInTheDocument();
})

test('Correct slide2 heading', () => {
  setup();
  const heading = screen.getByText("Product Details Can be viewed without Sign in")
  expect(heading).toBeInTheDocument();
})
test('Correct slide3 heading', () => {
  setup();
  const heading = screen.getByText("To add, edit, delete Please SignIn first")
  expect(heading).toBeInTheDocument();
})
test('Correct slide4 heading', () => {
  setup();
  const heading = screen.getByText("Products views and price analysis can be done through analysis")
  expect(heading).toBeInTheDocument();
})
})