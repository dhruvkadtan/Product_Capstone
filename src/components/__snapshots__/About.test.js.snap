// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render correctly 1`] = `
<div
  className="carousel slide"
  data-ride="carousel"
  id="carouselExampleCaptions"
>
  <div
    className="carousel-indicators"
  >
    <button
      aria-current="true"
      aria-label="Slide 1"
      className="active"
      data-bs-slide-to="0"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
    <button
      aria-label="Slide 2"
      data-bs-slide-to="1"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
    <button
      aria-label="Slide 3"
      data-bs-slide-to="2"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
    <button
      aria-label="Slide 4"
      data-bs-slide-to="3"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
  </div>
  <div
    className="carousel-inner"
  >
    <div
      className="carousel-item active "
    >
      <img
        alt="..."
        className="d-block  w-100"
        src="carousel1.jpg"
        style={
          Object {
            "filter": " blur(1px)",
          }
        }
      />
      <div
        className="carousel-caption d-none d-md-block "
      >
        <h3
          className="text-white fw-bold"
        >
          Welcome to D-Systems
        </h3>
        <h4
          className="text-white fw-bold"
        >
          Accounting and Inventory Software
        </h4>
        <h5>
          We provide a managing platform for businesses
        </h5>
        <h5>
             Here, we can account the stock products
        </h5>
        <h5>
              and also maintain daily transactions.
        </h5>
      </div>
    </div>
    <div
      className="carousel-item"
    >
      <img
        alt="..."
        className="d-block w-100"
        src="products.jpg"
      />
      <div
        className="carousel-caption d-none d-md-block"
      >
        <h5>
          Product Details Can be viewed without Sign in
        </h5>
      </div>
    </div>
    <div
      className="carousel-item"
    >
      <img
        alt="..."
        className="d-block w-100"
        src="signin.jpg"
      />
      <div
        className="carousel-caption d-none d-md-block"
      >
        <h5>
          To add, edit, delete Please SignIn first
        </h5>
      </div>
    </div>
    <div
      className="carousel-item"
    >
      <img
        alt="..."
        className="d-block w-100"
        src="analysis.jpg"
      />
      <div
        className="carousel-caption d-none d-md-block"
      >
        <h5>
          Products views and price analysis can be done through analysis
        </h5>
      </div>
    </div>
  </div>
</div>
`;
